### PR TITLE
cloudfoundry-cli: fix build on multiple platforms, notably darwin. also bump -> 6.41.0

### PR DESCRIPTION
**Pull request by risicle**
###### Motivation for this change

The `cloudfoundry-cli` package would previously build the linux/amd64 binary on all platforms. This was due to the projects makefiles having specific targets for different platforms.

I'm fairly certain this will also work on most platforms, it's just there isn't a specific makefile target for them and would require us to do some manual tinkering.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by risicle** [comment]
Note I've also pushed on top of this a bump from 6.37.0 to 6.41.0, but I want to keep them as separate commits so I can easily & clearly cherry-pick the build fix to stable.

**Comment by risicle** [comment]
I realize I've inadvertently tried to reinvent the `meta.platforms = ` mechanism here using the `disabled = ` attribute - am pushing a better version...

**Comment by Mic92** [comment]
Can you make the package install to $bin and remove `remove-references-to`?
buildGoPackage will then take care of removing go compiler references.

**Comment by risicle** [comment]
Think I've done what you want, but I'll note that I wasn't the one that added these things originally, despite what git annotate might tell you. I think I will be unticking "allow edits from maintainers" in future as it is so confusing.

Again, I'd appreciate it if you didn't squash these commits as I'll be looking to cherry pick the most straightforward platform-fixing commit for stable.

File: pkgs/development/tools/cloudfoundry-cli/default.nix [hunk 1]
```diff
@@ -12,26 +12,35 @@ buildGoPackage rec {
...
-  outputs = [ "out" ];
+  makeTarget = let hps = stdenv.hostPlatform.system; in
+    if hps == "x86_64-darwin" then
+      "out/cf-cli_osx"
+    else if hps == "x86_64-linux" then
+      "out/cf-cli_linux_x86-64"
+    else if hps == "i686-linux" then
+      "out/cf-cli_linux_i686"
+    else
+      abort "make target for this platform unknown";
```
[the remark this answers is not in the document]
Comment (risicle) [reply pkgs/development/tools/cloudfoundry-cli/default.nix]: ahok

**Pull request merged**